Assets/Scripts/InventoryAndItemsScripts/AreaThatChecksInventory.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Inventory
+{
+    [RequireComponent(typeof(InventoryChecker))]
+    [RequireComponent(typeof(Collider))]
+    public class AreaThatChecksInventory : MonoBehaviour
+    {
+        public event Action<GameObject> FailedInventoryCheck;
+        public event Action<GameObject> PassedInventoryCheck;
+
+        private void OnTriggerStay(Collider other)
+        {
+            if (other.gameObject.tag == "Player")
```

Comment:
invert check to reduce nesting
`if (other.gameObject.tag != "Player"){
    return;l
}`